src/Compilers/Core/CodeAnalysisTest/InternalUtilities/StreamExtensionsTests.cs

```diff
@@ -118,5 +118,33 @@ public void PrematureEndOfStream()
             var expected = new byte[] { 1, 2, 3, 4, 0, 0 };
             Assert.Equal(expected, destArray);
         }
+
+        [Fact]
+        public void ReadAllBytes_Seakable()
+        {
+            var sourceArray = new byte[] { 1, 2, 3, 4 };
+            var stream = new MemoryStream(sourceArray);
+            stream.ReadByte();
+            Assert.Equal(new byte[] { 2, 3, 4 }, stream.ReadAllBytes());
+        }
+
+        [Fact]
+        public void ReadAllBytes_NonSeekable()
+        {
+            var sourceArray = new byte[] { 1, 2, 3, 4 };
+            var stream = new TestStream(canSeek: false, backingStream: new MemoryStream(sourceArray));
+            stream.ReadByte();
+            Assert.Equal(new byte[] { 2, 3, 4 }, stream.ReadAllBytes());
+        }
+
+        [Fact]
+        public void ReadAllBytes_End()
+        {
+            var sourceArray = new byte[] { 1, 2 };
+            var stream = new TestStream(canSeek: false, backingStream: new MemoryStream(sourceArray));
+            stream.ReadByte();
+            stream.ReadByte();
+            Assert.Equal(new byte[0], stream.ReadAllBytes());
+        }
```

Comment:
Consider testing `ReadAllBytes()` when `TestStream.Length > backingStream.Length` (the `Array.Resize` case).